### PR TITLE
[Documentation] Logically grouping core CLI commands

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR logically groups Core CLI commands from the documentation. We will have additional refactoring post 2.0.

There are 4 sections

- Cluster Mangement
- Session
- Job
- State

As suggested https://docs.google.com/document/d/1_n7m2lSEA1clC9OARLxgm5jwpo4VxlqmqgYxosRrbic/edit#heading=h.wzsjlamcxbw9

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/package-ref.rst [hunk 1]
```diff
@@ -287,6 +291,16 @@ The Ray Command Line API
...
 
+.. _ray-monitor-doc:
+
+.. click:: ray.scripts.scripts:monitor
+   :prog: ray monitor
+   :show-nested:
+
+Session
+~~~~~~~
+This section contains commands for inspecting, debugging, and configuring the current cluster.
```
Comment (rkooo567) [inline doc/source/ray-core/package-ref.rst:302]: Not sure if this is the right explanation & right set of grouping. Please recommend me if you have a better alternative!

File: doc/source/ray-core/package-ref.rst [hunk 2]
```diff
@@ -332,3 +340,11 @@ The Ray Command Line API
 .. click:: ray.scripts.scripts:enable_usage_stats
    :prog: ray enable-usage-stats
    :show-nested:
+
+Job
+~~~
+CLIs to submit, stop, and monitor Ray jobs. See :ref:`Ray Job Submission API Reference <ray-job-submission-api-ref-under-construction>`.
```
Comment (rkooo567) [inline doc/source/ray-core/package-ref.rst:346]: Q: Should we add the actual reference here, or link should be sufficient?

**Review by rickyyx — COMMENTED** [review]
Should we actually promote the CLI API to one level higher with the others? 
image

The rest LGTM, I will leave to the doc committee to stamp.

**Comment by rkooo567** [comment]
link check unrelated. Also any comment from the @ray-project/ray-docs regarding @rickyyx's comment?

File: doc/source/ray-core/package-ref.rst (hunk 1 above)
Comment (richardliaw) [reply doc/source/ray-core/package-ref.rst:302]: what about "Running Clusters" or "Job"? session is not a defined concept

File: doc/source/ray-core/package-ref.rst (hunk 2 above)
Comment (richardliaw) [reply doc/source/ray-core/package-ref.rst:346]: hmm we should consolidate it here (not sure why its all over the place?)

File: doc/source/ray-core/package-ref.rst (hunk 1 above)
Comment (stephanie-wang) [reply doc/source/ray-core/package-ref.rst:302]: Seems like these are about "Debugging applications"?

File: doc/source/ray-core/package-ref.rst [hunk 3]
```diff
@@ -239,6 +239,10 @@ Debugging APIs
 The Ray Command Line API
 ------------------------
 
+Cluster Management
```
Comment (jjyao) [inline doc/source/ray-core/package-ref.rst:242]: I thought we are going to group commands printed by `ray --help`?

Comment (ericl) [reply doc/source/ray-core/package-ref.rst:242]: Yeah, you should be able to define command groups there in click I believe.

File: doc/source/ray-core/package-ref.rst (hunk 1 above)
Comment (rkooo567) [reply doc/source/ray-core/package-ref.rst:302]: Usage stats are technically not in there. Maybe I can make a new section?

Comment (rkooo567) [reply doc/source/ray-core/package-ref.rst:302]: I used `Debugging applications` and made a new section for usage stats

File: doc/source/ray-core/package-ref.rst (hunk 3 above)
Comment (rkooo567) [reply doc/source/ray-core/package-ref.rst:242]: I did some research, and it is a bit difficult (there's no public API to customize help messages) https://stackoverflow.com/questions/62182687/custom-help-in-python-click

Comment (rkooo567) [reply doc/source/ray-core/package-ref.rst:242]: Should we do this in this PR? I am planning to cherry pick for 2.0, but this change might be a bit risky

File: doc/source/ray-core/package-ref.rst (hunk 2 above)
Comment (rkooo567) [reply doc/source/ray-core/package-ref.rst]: Hmm not sure why. For state APIs, I just followed job API (also these 2 features have lots of APIs, and I think it is a bit weird to spread them under core API page).

**Comment by rkooo567** [comment]
I may replace the PR to https://github.com/ray-project/ray/pull/27862. Need Jules feedback before that

**Pull request closed without merging**